### PR TITLE
#2500, #2507

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
#2500 REPL stuck spinning when trying to show help
#2507 Remote: rtvs::: is not autocompleting

### CODE REVIEW

File: src/Host/Client/Impl/Session/RSession.cs [hunk 1]
```diff
@@ -687,10 +687,12 @@ internal sealed class RSession : IRSession, IRCallbacks {
         /// <param name="url"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        Task IRCallbacks.WebBrowser(string url, CancellationToken cancellationToken) {
-            var newUrl = BrokerClient.HandleUrl(url, cancellationToken);
+        async Task IRCallbacks.WebBrowser(string url, CancellationToken cancellationToken) {
+            var newUrl = await BrokerClient.HandleUrlAsync(url, cancellationToken);
             var callback = _callback;
-            return callback != null ? callback.ShowHelpAsync(newUrl, cancellationToken) : Task.CompletedTask;
+            if (callback != null) {
```
Comment (AlexanderSher) [inline src/Host/Client/Impl/Session/RSession.cs]: We can check `_callback` before calling `HandleUrlAsync`

File: src/Host/Client/Impl/BrokerServices/WebServer.cs [hunk 2]
```diff
@@ -11,41 +11,51 @@
...
-                _listener.Prefixes.Add(Invariant($"http://{LocalHost}:{LocalPort}/"));
-                try {
-                    _listener.Start();
-                } catch (HttpListenerException) {
-                    _listener.Close();
-                    continue;
+        }
+
+        public Task InitializeAsync(CancellationToken ct) {
+            return Task.Run(() => {
```
Comment (AlexanderSher) [inline src/Host/Client/Impl/BrokerServices/WebServer.cs]: Can you explain this change?

Comment (MikhailArkhipov) [inline src/Host/Client/Impl/BrokerServices/WebServer.cs]: #2500. Original code did the loop in sync without option to cancel. From the dump it appears loop was stuck in HttpListener.AddPrefix.

**Comment by AlexanderSher** [comment]
👍

**Pull request merged**